backend/handler/handler.go

```diff
@@ -291,15 +357,26 @@ func (h *Handler) Sell(c echo.Context) error {
 
 	item, err := h.ItemRepo.GetItem(ctx, req.ItemID)
 	// TODO: not found handling
-	// http.StatusPreconditionFailed(412)
+	// http.StatusNotFound(404)
 	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return echo.NewHTTPError(http.StatusNotFound, "Item not found")
+		}
 		return echo.NewHTTPError(http.StatusInternalServerError, err)
 	}
+	// no need to check user 404 because it is loggged in already
 
 	// TODO: check req.UserID and item.UserID
 	// http.StatusPreconditionFailed(412)
+	if req.UserID != item.UserID {
+		return echo.NewHTTPError(http.StatusPreconditionFailed, "the item is not yours")
+	}
+
 	// TODO: only update when status is initial
 	// http.StatusPreconditionFailed(412)
+	if item.Status != domain.ItemStatusInitial {
+		return echo.NewHTTPError(http.StatusPreconditionFailed, "the item is in the state of initial. It has been on sale or sold.")
```

Comment:
Im not sure what this error message means, do you mind clarifying? (you dont have to update the code, just responding to my comment is fine)

Reply:
the add-items and sell are two separate functions. successfully adding items into the db doesn't mean it is on sale. Therefore, when selling an item, we want to make sure only the item that is pending to be sold can move forward.

Reply:
I see, so essentially this error is for when someone tries to sell an item that has already been listed?